### PR TITLE
Audit fix 6: unbounded iteration in PerformSubmitTx()

### PR DESCRIPTION
**Pull request by foxpy**
Tracking task: [NTRN-217](https://p2pvalidator.atlassian.net/browse/NTRN-217).

### CODE REVIEW

**Review by zavgorodnii — APPROVED** [review]

File: wasmbinding/message_plugin.go [hunk 1]
```diff
@@ -201,6 +201,20 @@ func (m *CustomMessenger) PerformSubmitTx(ctx sdk.Context, contractAddr sdk.AccA
 		InterchainAccountId: submitTx.InterchainAccountId,
 		Timeout:             submitTx.Timeout,
 	}
+	params := m.Keeper.GetParams(ctx)
+	if uint64(len(submitTx.Msgs)) > params.GetMsgSubmitTxMaxMessages() {
```
Comment (swelf19) [inline wasmbinding/message_plugin.go]: you validate only the wasm bindings. But user(contract) still can craft the proto(stargate) msg and ignore the limitation. am i right?

Comment (foxpy) [reply wasmbinding/message_plugin.go]: I have no idea.

CC @pr0n00gler @oopcode

**Comment by foxpy** [comment]
Actually, this loop might be removed entirely. I will open this PR again when I finish investigating.

File: wasmbinding/message_plugin.go (hunk 1 above)
Comment (foxpy) [reply wasmbinding/message_plugin.go]: Moved validation from `PerformSubmitTx()` to keeper's `SubmitTx()`

**Comment by foxpy** [comment]
> Actually, this loop might be removed entirely.

We cannot get rid of this loop since protobuf's `Any` uses custom JSON (un)marshalling logic and we heavily rely on (un)marshalling it.

**Review by pr0n00gler — APPROVED** [review]

**Review by swelf19 — APPROVED** [review]

**Pull request merged**